packages/db/index.js

```diff
@@ -1295,6 +1295,25 @@ export class DBClient {
     }
   }
 
+  /**
+   * @param {string} userId
+   * @param {import('./db-client-types').AgreementKind} agreement
+   * @returns {Promise<void>}
+   */
+  async createUserAgreement (userId, agreement) {
+    const { error } = await this._client
+      .from('agreement')
+      .insert({
```

Comment:
As an insert, won't `error` be populated if the same user adds the same agreement?  Then this will throw and it doesn't seem like it's being caught.  It's possible this intended (also possible I'm misreading), but just wanted to call it out just in case.

Everything else looks great!  I love the solution.

Reply:
I got rid of the uniqueness constraint in tables and in the migration. So... I think it won't error in that case eh?
here is the relevant test: https://github.com/web3-storage/web3.storage/pull/2028/commits/5c84b5bb82279e876d8ca5f47b3497a0ee2482c5

Or am I misunderstanding?